packages/salesforcedx-vscode-core/src/metadataSupport/metadataDocumentationService.ts

```diff
@@ -0,0 +1,720 @@
+  description: string;
+  required?: boolean;
+  validValues?: string[];
+};
+
+export type MetadataTypeDocumentation = {
+  name: string;
+  description: string;
+  fields?: MetadataFieldInfo[];
+  developerGuideUrls?: string[];
+};
+
+/**
+ * Service for loading and providing metadata type documentation
+ */
+export class MetadataDocumentationService {
+  private documentationMap: Map<string, MetadataTypeDocumentation> = new Map();
+  private initialized = false;
+
+  /**
+   * Initialize the service by loading metadata documentation from XSD
+   */
+  public async initialize(): Promise<void> {
+    if (this.initialized) {
+      return;
+    }
+
+    try {
+      await this.loadMetadataDocumentation();
+      this.initialized = true;
+    } catch (error) {
+      console.error('Failed to initialize metadata documentation service:', error);
+    }
+  }
+
+  /**
+   * Get documentation for a specific metadata type
+   */
+  public getDocumentation(metadataType: string): MetadataTypeDocumentation | null {
+    return this.documentationMap.get(metadataType) ?? null;
+  }
+
+  /**
+   * Get documentation for a specific field within a metadata type
+   */
+  public getFieldDocumentation(metadataType: string, fieldName: string): Promise<MetadataFieldDocumentation | null> {
+    // Get field documentation from our hardcoded field definitions
+    const fieldDoc = this.getFieldDefinitions(metadataType, fieldName);
+    if (fieldDoc) {
+      return Promise.resolve(fieldDoc);
+    }
+
+    // If no hardcoded definition, try to extract from XSD patterns
+    return Promise.resolve(this.extractFieldFromXSDPatterns(metadataType, fieldName));
+  }
+
+  /**
+   * Get all available metadata types
+   */
+  public getAllMetadataTypes(): string[] {
+    return Array.from(this.documentationMap.keys()).sort();
+  }
+
+  /**
+   * Load metadata documentation from the official Salesforce Developer Guide
+   * This references the authoritative documentation at:
+   * https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_types_list.htm
+   */
+  private async loadMetadataDocumentation(): Promise<void> {
+    // Load comprehensive metadata type documentation from the official Salesforce Developer Guide
+    // This is the authoritative source of truth for all metadata types
+    const metadataTypes = this.getOfficialMetadataTypes();
+
+    for (const [typeName, typeInfo] of Object.entries(metadataTypes)) {
+      this.documentationMap.set(typeName, {
+        name: typeName,
+        description: typeInfo.description ?? '',
+        fields: typeInfo.fields,
+        developerGuideUrls: this.getDeveloperGuideUrls(typeName)
+      });
+    }
+  }
+
+  /**
+   * Get official metadata types from the Salesforce Metadata API Developer Guide
+   * Source: https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_types_list.htm
+   */
+  private getOfficialMetadataTypes(): Record<string, Partial<MetadataTypeDocumentation>> {
```

Comment:
why only these?  what's "official" about them vs. the others?

definitely alphabetize so they match the docs (which are alphabetized)

Reply:
this should probably be its own file, and somewhere in this PR should be the code/script that turns the docs page in your `Source:` link into this programatically for the next time it needs updating.

[which we could do, programmatically...ex: that's how SDR maintains https://github.com/forcedotcom/source-deploy-retrieve/blob/main/METADATA_SUPPORT.md]